### PR TITLE
[AIRFLOW-3561] Improve queries

### PR DESCRIPTION
**Pull request by ffinfo**
Make sure you have checked _all_ steps below.

### Jira

- [x] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "\[AIRFLOW-3561\] My Airflow PR"
  - https://issues.apache.org/jira/browse/AIRFLOW-3561

### Description

This pull request will reduce interaction with the DagBag en improves database queries.

### Tests

- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

### Commits

- [x] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [x] In case of new functionality, my PR adds documentation that describes how to use it.
  - When adding new operators/hooks/sensors, the autoclass documentation generation needs to be added.
  - All the public functions and the classes in the PR contain docstrings that explain what it does

### Code Quality

- [x] Passes `flake8`

### CODE REVIEW

File: airflow/models/__init__.py [hunk 1]
```diff
@@ -3249,6 +3259,17 @@ def __exit__(self, _type, _value, _tb):
...
 
+    @property
+    def default_view(self):
+        if self._default_view is None:
+            return configuration.conf.get('webserver', 'dag_default_view').lower()
+        else:
+            return self._default_view
+
+    def get_default_view(self):
+        """This is only there for backward compatible jina2 templates"""
```
Comment (jmcarp) [inline airflow/models/__init__.py]: I think I'm missing something here. Why can't the templates continue using the `default_view` property? Also, s/jina2/jinja2/.

File: airflow/models/__init__.py [hunk 2]
```diff
@@ -4201,6 +4222,8 @@ def sync_to_db(self, owner=None, sync_time=None, session=None):
         orm_dag.owners = owner
         orm_dag.is_active = True
         orm_dag.last_scheduler_run = sync_time
+        orm_dag.default_view = self._default_view
```
Comment (jmcarp) [inline airflow/models/__init__.py:4225]: Can we test this in `tests.models:DagTest.test_sync_to_db`?

File: airflow/models/__init__.py [hunk 3]
```diff
@@ -3249,6 +3259,17 @@ def __exit__(self, _type, _value, _tb):
 
     # /Context Manager ----------------------------------------------
 
+    @property
+    def default_view(self):
```
Comment (jmcarp) [inline airflow/models/__init__.py]: Can we test this property?

**Comment by jmcarp** [comment]
Until #4339 is merged, we have to make duplicate changes to `airflow/www/views` and `airflow/www_rbac/views`.

File: airflow/models/__init__.py (hunk 1 above)
Comment (ffinfo) [reply airflow/models/__init__.py]: This was to do with https://issues.apache.org/jira/browse/AIRFLOW-3562.
Currently DAG is still used in other templates, can't change all those at once. Once all views has switched to DagModel this method can be removed again.

Comment (ffinfo) [reply airflow/models/__init__.py]: ps: fixing typo ;)

File: airflow/models/__init__.py (hunk 2 above)
Comment (ffinfo) [reply airflow/models/__init__.py:4225]: Good one, added a test

File: airflow/models/__init__.py (hunk 3 above)
Comment (ffinfo) [reply airflow/models/__init__.py]: Good one, added a test

Comment (Fokko) [reply airflow/models/__init__.py]: This behavior how differs from the `get_default_view` method. Maybe we can remove the `get_default_view` method completely: https://stackoverflow.com/questions/49134141/how-do-i-reference-an-object-property-in-a-method-call-in-jinja2

This would reduce the number of methods on the DAG object. Besides that, I believe that `dag.default_view` should return the same as `dag.get_default_view()` since the names are so similar.

File: airflow/www/views.py [hunk 4]
```diff
@@ -550,31 +550,29 @@ def chart(self):
...
-        ds.update(
-            dag_ids=[dag.dag_id for dag in dagbag.dags.values() if not dag.is_subdag]
-        )
-
-        qry = (
-            session.query(ds.dag_id, ds.state, ds.count)
-        )
+        states = session.query(dr.dag_id, dr.state, sqla.func.count(dr.state)).group_by(dr.dag_id, dr.state)
 
         data = {}
```
Comment (Fokko) [inline airflow/www/views.py:559]: Maybe give this a bit more meaningful name? Something like `dag_state_stats`.

**Review by Fokko — CHANGES_REQUESTED** [review]
Small suggestions, apart from that it looks great!

**Comment by Fokko** [comment]
Two related CI issues:
```

======================================================================
3) ERROR: test_sync_to_db (tests.DagTest)
----------------------------------------------------------------------
   Traceback (most recent call last):
    .tox/py27-backend_mysql-env_docker/lib/python2.7/site-packages/mock/mock.py line 1305 in patched
      return func(*args, **keywargs)
    tests/models.py line 749 in test_sync_to_db
      self.assertIsNone(orm_dag._default_view)
   AttributeError: 'DagModel' object has no attribute '_default_view'
   -------------------- >> begin captured stdout << ---------------------
   [2018-12-26 08:08:11,955] {__init__.py:4219} INFO - Creating ORM DAG for dag
   [2018-12-26 08:08:11,963] {__init__.py:4219} INFO - Creating ORM DAG for dag.subtask
   
   --------------------- >> end captured stdout << ----------------------
   -------------------- >> begin captured logging << --------------------
   airflow.models.DAG: INFO: Creating ORM DAG for dag
   airflow.models.DAG: INFO: Creating ORM DAG for dag.subtask
   --------------------- >> end captured logging << ---------------------
======================================================================
4) ERROR: test_sync_to_db_default_view (tests.DagTest)
----------------------------------------------------------------------
   Traceback (most recent call last):
    .tox/py27-backend_mysql-env_docker/lib/python2.7/site-packages/mock/mock.py line 1305 in patched
      return func(*args, **keywargs)
    tests/models.py line 781 in test_sync_to_db_default_view
      self.assertIsNotNone(orm_dag._default_view)
   AttributeError: 'DagModel' object has no attribute '_default_view'
```

File: airflow/models/__init__.py (hunk 3 above)
Comment (ffinfo) [reply airflow/models/__init__.py]: @Fokko Agree there. I did now remove all usage of dag.default_view and switch it to dag.get_default_view()

Comment (ffinfo) [reply airflow/models/__init__.py]: @Fokko: Can't use just `default_view` because in DagModel this can be None and get_default_view has a fall back to the config when not set.

File: airflow/www/views.py (hunk 4 above)
Comment (ffinfo) [reply airflow/www/views.py:559]: 👍

**Comment by ffinfo** [comment]
Rbac is now also fixed

File: airflow/models/__init__.py (hunk 3 above)
Comment (Fokko) [reply airflow/models/__init__.py]: Awesome, thanks!

**Review by Fokko — APPROVED** [review]

File: airflow/models/__init__.py [hunk 5]
```diff
@@ -2999,15 +2999,29 @@ class DagModel(Base):
...
+    description = Column(Text)
+    # Default view of the inside the webserver
+    default_view = Column(String(25))
 
     def __repr__(self):
         return "<DAG: {self.dag_id}>".format(self=self)
 
+    @property
+    def timezone(self):
+        return settings.TIMEZONE
```
Comment (ashb) [inline airflow/models/__init__.py:3012]: Wait up - this looks incorrect. Why was this removed?

The timezone of a DAG was previously defined as the timezone of the start_date. Is that still the case and if we do `self.timezone = tz` then this property accessor won't be used anymore?

Comment (ashb) [reply airflow/models/__init__.py:3012]: I don't  mean removed, I mean added.

Comment (ashb) [reply airflow/models/__init__.py:3012]: Oh this is on the DagModel, not the DAG. Right.

In which case my question is: when/where is this used?

Comment (ffinfo) [reply airflow/models/__init__.py:3012]: This is indeed a bit tricky. I have added this because in the html templates this variable is used. Some templates now takes a DagModel instead of a DAG. In the end I would like this to be in the complete webserver

**Comment by ashb** [comment]
We reverted this from master? Did it ever make it back in in any form?

**Comment by ffinfo** [comment]
@ashb Seems to be still in there:
https://github.com/apache/airflow/blame/master/airflow/www/views.py#L299

Because of the switch to the rbac interface it's was moved I think
